frontend/src/components/homepage/PreviewModal.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from "react";
+import { gql, useQuery } from "@apollo/client";
+import {
+  Badge,
+  Box,
+  Button,
+  Heading,
+  Link,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalCloseButton,
+  Text,
+} from "@chakra-ui/react";
+import convertLanguageTitleCase from "../../utils/LanguageUtils";
+
+export type PreviewModalProps = {
+  storyId: number;
+  title: string;
+  youtubeLink: string;
+  level: number;
+  language: string;
+  previewBook: () => void;
+  preview: boolean;
+  primaryBtnText: string;
+  primaryBtnOnClick: () => any;
+};
+
+const PreviewModal = ({
+  storyId,
+  title,
+  youtubeLink,
+  level,
+  language,
+  previewBook,
+  preview,
+  primaryBtnText,
+  primaryBtnOnClick,
+}: PreviewModalProps) => {
+  const [content, setContent] = useState<string[]>([]);
+
+  const GET_STORY_CONTENTS = (id: number) =>
+    gql`
+      query {
+        storyById(
+          id: ${id}
+        ) {
+          contents {
+            id
+            lineIndex
+            content
+          }
+        }
+      }
+    `;
+
+  useQuery(GET_STORY_CONTENTS(storyId), {
+    fetchPolicy: "cache-and-network",
+    onCompleted: (data) => {
+      const storyContentsArray = data.storyById.contents.map((story: any) => {
+        return story.content;
+      });
+      setContent(storyContentsArray);
+    },
+  });
+
+  const storyContents = content.map((c: string, index: number) => (
+    <Box display="flex" padding={3}>
```

Comment:
` <Box display="flex"> === <Flex>`

Reply:
also i have a feeling if you use `<Flex direction="row">` you wont need the width on boxes